src/include/daos_fs_sys.h

```diff
@@ -0,0 +1,454 @@
+typedef struct dfs_sys {
+	dfs_t			*dfs;		/* mounted filesystem */
+	struct d_hash_table	*dfs_hash;	/* optional lookup hash */
+} dfs_sys_t;
+
+/**
+ * Mount a file system with dfs_mount and optionally
+ * initialize a cache.
+ *
+ * \param[in]	poh	Pool connection handle.
+ * \param[in]	coh	Container open handle.
+ * \param[in]	mflags	Mount flags (O_RDONLY or O_RDWR).
+ * \param[in]	sflags	Sys flags (DFS_SYS_NO_CACHE or DFS_SYS_NO_LOCK)
+ * \param[out]	dfs_sys	Pointer to the file system object created.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_mount(daos_handle_t poh, daos_handle_t coh, int mflags, int sflags,
+	      dfs_sys_t **dfs_sys);
+
+/**
+ * Unmount a file system with dfs_mount.
+ *
+ * \param[in]	dfs_sys	Pointer to the mounted file system.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_umount(dfs_sys_t *dfs_sys);
+
+/**
+ * Check access permissions on a path. Similar to Linux access(2).
+ * By default, symlinks are dereferenced.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in]	mask	accessibility check(s) to be performed.
+ *			It should be either the value F_OK, or a mask with
+ *			bitwise OR of one or more of R_OK, W_OK, and X_OK.
+ * \param[in]	flags	Access flags (O_NOFOLLOW).
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_access(dfs_sys_t *dfs_sys, const char *path, int mask, int flags);
+
+/**
+ * Change permission access bits. Symlinks are dereferenced.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in]	mode	New permission access modes. For now, we don't support
+ *			the sticky bit, setuid, and setgid.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_chmod(dfs_sys_t *dfs_sys, const char *path, mode_t mode);
+
+/**
+ * set stat attributes for a file and fetch new values.
+ * By default, if the object is a symlink the link itself is modified.
+ * See dfs_sys_stat() for which entries are filled.
+ *
+ * \param[in]	dfs_sys	Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in,out]
+ *		stbuf	[in]: Stat struct with the members set.
+ *			[out]: Stat struct with all valid members filled.
+ * \param[in]	flags	Bitmask of flags to set.
+ * \param[in]	sflags	(O_NOFOLLOW)
+ *
+ * \return		0 on Success. errno code on Failure.
+ */
+int
+dfs_sys_setattr(dfs_sys_t *dfs_sys, const char *path, struct stat *stbuf,
+		int flags, int sflags);
+
+/**
+ * Set atime and mtime of a path.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	path	Link path of object.
+ * \param[in]	times	[0]: atime to set
+ *			[1]: mtime to set
+ * \param[in]	flags	(O_NOFOLLOW)
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_utimens(dfs_sys_t *dfs_sys, const char *path,
+		const struct timespec times[2], int flags);
```

Comment:
Alignment issues on several of these.

Reply:
I'm seeing this on GitHub as well, but when I go to the file, it seems to be okay. It looks like GitHub is rendering the tabs here as 6 spaces instead of 8. (for me at least)

Reply:
yea i have seen the same issue as well on my PRs